cosmwasm/packages/registry/src/registry.rs

```diff
@@ -323,66 +344,121 @@ impl<'a> Registry<'a> {
+            return Err(RegistryError::InvalidDenomTracePath {
+                path: format!("{path:?}"),
+                denom: coin.denom,
+            });
+        };
 
-        // default the receiver chain to the first chain if it isn't provided
+        // default the receiver chain to the current chain if it isn't provided
         let receiver_chain = match into_chain {
             Some(chain) => chain,
-            None => first_chain.as_ref(),
+            None => current_chain.as_ref(),
         };
+        // normalize it
         let receiver_chain: &str = &receiver_chain.to_lowercase();
 
         // If the token we're sending is native, we need the receiver to be
         // different than the origin chain. Otherwise, we will try to make an
         // ibc transfer to the same chain and it will fail. This may be possible
         // in the future when we have IBC localhost channels
-        if first_channel.is_none() && first_chain.as_ref() == receiver_chain {
+        if coin_type.is_native() && current_chain.as_ref() == receiver_chain {
             return Err(RegistryError::InvalidHopSameChain {
                 chain: receiver_chain.into(),
             });
         }
 
         // validate the receiver matches the chain
         let receiver_prefix = self.get_bech32_prefix(receiver_chain)?;
-        if receiver[..receiver_prefix.len()] != receiver_prefix {
+        if receiver_addr[..receiver_prefix.len()] != receiver_prefix {
             return Err(RegistryError::InvalidReceiverPrefix {
-                receiver,
+                receiver: receiver_addr,
                 chain: receiver_chain.into(),
             });
         }
 
-        let ts = block_time.plus_seconds(PACKET_LIFETIME);
+        // Define the data to be used in the IBC MsgTransfer.
+
+        // If the coin is native, the ibc transfer needs to be sent to the
+        // receiver. Otherwise, it needs to be sent to the channel we received
+        // it from
+        let first_transfer_chain = match coin_type {
+            CoinType::Native => receiver_chain,
+            CoinType::Ibc => path[1].on.as_ref(), // Non native coins always have a path of len > 1 per definition
+        };
+
+        // encode the receiver address for the first chain
+        let first_receiver = self.encode_addr_for_chain(&receiver_addr, first_transfer_chain)?;
+        let first_channel = match first_channel {
+            Some(channel) => Ok::<String, RegistryError>(channel.as_ref().to_string()),
+            None => {
+                // If there is no first channel (i.e. coin_type.is_native() ==
+                // True), we get the channel from the registry
+                assert!(coin_type.is_native());
+                assert!(first_transfer_chain == receiver_chain);
+                let channel = self.get_channel(current_chain.as_ref(), first_transfer_chain)?;
+                Ok(channel)
+            }
+        }?;
```

Comment:
Feels simpler to me to just not run this code and avoid `!native` checks throughout